ckanext/datastore_profiler/tests/test_datestatistics.py

```diff
@@ -1,74 +1,154 @@
-# User defined modules
+"""
+Test module for datestatistics module using pytest
+
+Test date_count and date_or_timestamp methods under DateStatistics class.
+The module contains the following functions:
+- test_date_format_error
+- test_date_min
+- test_date_max
+- test_null_count
+- test_yearmonth_count
+- test_year_count
+- test_weekday_count
+- test_hour_count
+"""
+
+import pytest
 from utils.datestatistics import DateStatistics
-from datetime import datetime
 
-test_dates = ["2020-06-12", "2020-06-12", "2001-06-12", "2020-06-21", "2020-06-26", "2020-06-26", "1987-11-12", "2020-11-12", "2020-11-12", "2020-11-12", "2020-12-12", None, None]
-test_timestamps = ["2020-06-12T07:30:00", "2020-06-12T07:30:00", "2001-06-12T16:30:00", "2020-06-21T16:30:00", "2020-06-26T16:30:00", "2020-06-26T07:30:00", "1987-11-12T07:15:00", "2020-11-12T07:15:00", "2020-11-12T07:15:00", "2020-11-12T07:30:22", "2020-12-12T07:30:22", None, None]
-
-date_dict_stats = DateStatistics().date_count(input = test_dates)
-timestamp_dict_stats = DateStatistics().date_count(input = test_timestamps)
-
-correct_min = '1987-11-12'
-correct_max = '2020-12-12'
+test_dates = [
+    "2020-06-12",
+    "2020-06-12",
+    "2001-06-12",
+    "2020-06-21",
+    "2020-06-26",
+    "2020-06-26",
+    "1987-11-12",
+    "2020-11-12",
+    "2020-11-12",
+    "2020-11-12",
+    "2020-12-12",
+    None,
+    None,
+]
+test_timestamps = [
+    "2020-06-12T07:30:00",
+    "2020-06-12T07:30:00",
+    "2001-06-12T16:30:00",
+    "2020-06-21T16:30:00",
+    "2020-06-26T16:30:00",
+    "2020-06-26T07:30:00",
+    "1987-11-12T07:15:00",
+    "2020-11-12T07:15:00",
+    "2020-11-12T07:15:00",
+    "2020-11-12T07:30:22",
+    "2020-12-12T07:30:22",
+    None,
+    None,
+]
+input_mix = ["2020-06-12 07:30:00", "2020-01-01"]
+
+date_dict_stats = DateStatistics().date_count(input=test_dates)
+timestamp_dict_stats = DateStatistics().date_count(input=test_timestamps)
+
+correct_min = "1987-11-12"
+correct_max = "2020-12-12"
 correct_null_count = 2
-correct_yearmonth_count = {'2020-June': 5, '2020-November': 3, '2001-June': 1, '1987-November': 1, '2020-December': 1}
-correct_year_count = {'2020': 9, '2001': 1, '1987': 1, }
-correct_hour_count = {'07': 8, '16': 3}
-correct_weekday_count = {'Friday': 4, 'Saturday': 1, 'Sunday': 1, 'Thursday': 4, 'Tuesday': 1}
+correct_yearmonth_count = {
+    "2020-June": 5,
+    "2020-November": 3,
+    "2001-June": 1,
+    "1987-November": 1,
+    "2020-December": 1,
+}
+correct_year_count = {
+    "2020": 9,
+    "2001": 1,
+    "1987": 1,
+}
+correct_hour_count = {"07": 8, "16": 3}
+correct_weekday_count = {
+    "Friday": 4,
+    "Saturday": 1,
+    "Sunday": 1,
+    "Thursday": 4,
+    "Tuesday": 1,
+}
+
+
+def test_date_format_error():
```

Comment:
this is useful - we dont have any other `pytest.raises` statements in our other tests, so I'm glad to see a working example